File: 프로그래머스/lv2/김미래/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <string>
#include <iostream>
using namespace std;
bool compare(string num1, string num2){

    return num1+num2>num2+num1;
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> strNum;
  
    for(int i=0; i<numbers.size(); i++){
        strNum.push_back(to_string(numbers[i]));
    }
    
    sort(strNum.begin(), strNum.end(), compare);
    
    for(int i=0; i<strNum.size(); i++){
        answer += strNum[i];
    }
    
    if(strNum[0]=="0"){
       answer = "0";
    }
    
    return answer;
}

File: 프로그래머스/lv2/임강민/가장 큰 수.py
def solution(numbers):
    numbers = list(map(str, numbers))
    # 문자열로 정렬하면 맨 앞 부터 비교 -> 3배 곱해줘서 뒤에 까지 정렬해주기
    numbers.sort(key=lambda x: x * 3, reverse=True)
    answer = str(int(''.join(numbers)))

    return answer